src/string/utf16.cr

```diff
@@ -12,22 +12,30 @@ class String
   # "hi 𐂥".to_utf16 # => Slice[104_u16, 105_u16, 32_u16, 55296_u16, 56485_u16]
   # ```
   def to_utf16 : Slice(UInt16)
-    size = 0
+    if ascii_only?
+      # size == bytesize, so each char fits in one UInt16
+      return to_slice.map &.to_u16
```

Comment:
Oh, actually, this is missing the trailing null. I totally forgot about that! Sorry, maybe the other way was fine, or we just need to copy the values but add a trailing null somehow.

Reply:
Also, are there specs for these cases? They should have failed with this.

Reply:
Oh, right. There should be a spec for that.